crates/ide/src/references.rs

```diff
@@ -35,44 +33,26 @@ pub struct ReferenceSearchResult {
 
 #[derive(Debug, Clone)]
 pub struct Declaration {
-    pub nav: NavigationTarget,
-    pub kind: ReferenceKind,
-    pub access: Option<ReferenceAccess>,
+    nav: NavigationTarget,
+    name: ast::NameLike,
+    access: Option<ReferenceAccess>,
```

Comment:
This actually feels like a step in the wrong direction. Types in `ide` should generally be  PODs with all fields public, and they should not carry ast/hir inside. Roughly, it should be OK to slap `derive(Serialize)` on them

Reply:
So is `FileReference` containing `NameLike` also a problem then? It is defined in `ide_db` but re-exported from `ide`.

Reply:
yeah, the fact that it is re-exported is problematic